WordPress/Classes/ViewRelated/NUX/LoginSocialErrorViewController.swift

```diff
@@ -0,0 +1,153 @@
+import Foundation
+import Gridicons
+
+class LoginSocialErrorViewController: UITableViewController {
+    fileprivate var errorTitle: String
+    fileprivate var errorDescription: String
+
+    fileprivate enum Sections: Int {
+        case titleAndDescription = 0
+        case buttons = 1
+
+        static var count: Int {
+            return buttons.rawValue + 1
+        }
+    }
+
+    fileprivate enum Buttons: Int {
+        case tryEmail = 0
+        case tryAddress = 1
+        case signup = 2
+    }
+
+    init(title: String, description: String) {
+        errorTitle = title
+        errorDescription = description
+
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        errorTitle = aDecoder.value(forKey: "errorTitle") as? String ?? ""
+        errorDescription = aDecoder.value(forKey: "errorDescription") as? String ?? ""
+
+        super.init(coder: aDecoder)
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+
+        tableView.estimatedRowHeight = 100.0
+
+        view.backgroundColor = WPStyleGuide.greyLighten30()
+    }
+
+    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        guard indexPath.section == Sections.buttons.rawValue else {
+            return
+        }
+
+        let controllerKey: String
+        switch indexPath.row {
+        case Buttons.tryEmail.rawValue:
+            controllerKey = "emailEntry"
+        case Buttons.tryAddress.rawValue:
+            controllerKey = "siteAddress"
+        case Buttons.signup.rawValue:
+            fallthrough
+        default:
+            controllerKey = "SignupViewController"
+        }
+
+        let storyboard = UIStoryboard(name: "Login", bundle: nil)
+        if let controller = storyboard.instantiateViewController(withIdentifier: controllerKey) as? NUXAbstractViewController {
+            navigationController?.setViewControllers([controller], animated: true)
+        }
+    }
+}
+
+
+// MARK: UITableViewDelegate methods
+
+extension LoginSocialErrorViewController {
+    override func tableView(_ tableView: UITableView, estimatedHeightForRowAt indexPath: IndexPath) -> CGFloat {
```

Comment:
I think this method isn't needed when setting `tableView.estimatedRowHeight`

Reply:
Ah damn, right.